src/test/java/test/DynamicGraphTest.java

```diff
@@ -74,15 +73,19 @@ public void test8() {
     dg.addEdge(0, c1, x, y);
     assertFreeNodesEquals(dg, y, x);
     dg.setStatus( dg.getFreeNodes(), Status.RUNNING);
-
-    assertFreeNodesEquals(dg, a1, a2);
+    assertFreeNodesEquals(dg);
```

Comment:
What I don't understand is why `a2` was a free node before because there is an edge from `a2` to `a1` and `a1` is not yet running.